ServiceAccounts: Add secret scan service docs (#57926)
grafana/grafana: docs/sources/setup-grafana/configure-security/secretscan.md

---

> aliases:

Reviewer: @Jguer, is this a new file (it appears to be) - if so, we don't need aliases. We only add an alias when the topic moves.

  - /docs/grafana/latest/auth/secretscan/
  - /docs/grafana/latest/setup-grafana/configure-security/secretscan/
description: Detect and revoke leaked Grafana service account tokens

> title: Grafana Secret Scanning

Reviewer: ```suggestion
title: Configure Grafana secret scanning
menuTitle: Configure secret scanning
```

weight: 1000
---

# Grafana Secret Scanning

> Grafana allows you to check if your [Service account tokens]({{< relref "../../administration/service-accounts/" >}})
> are leaked in GitHub through the [GitHub Secret Scanning](https://docs.github.com/en/code-security/secret-scanning/about-secret-scanning) service.

Reviewer: ```suggestion
With Grafana, you can use the GitHub Secret Scanning service to determine if your [service account tokens]({{< relref "../../administration/service-accounts/" >}}) have been leaked on GitHub.
```

<div class="clearfix"></div>

> ## Token Revocation

Reviewer: I don't think a H2 is necessary here. The content in this section just continues what you've written above. Let's remove it.

> ## Token Revocation

Reviewer: ```suggestion
```

> When a Grafana secret is detected by GitHub Secret Scanning, its hash is stored in Grafana's secret scanning service.

Reviewer: ```suggestion
When GitHub Secret Scanning detects a Grafana secret, its hash is stored in Grafana's secret scanning service.
```

> On-premises and cloud Grafana instances can query this service to check if a token emitted by the instance has been publicly exposed by
> hash-to-hash comparison.

Reviewer: ```suggestion
Grafana instances, whether on-premises or on the cloud, can use this service to verify if a token generated by the instance has been made public. This verification is done by comparing the token's hash with the exposed token's hash.
```

When a token is found to be leaked, the service will automatically revoke the token, rendering it unusable.

> > **Note:** if the `revoke` option is disabled, the service will only send a notification to the configured webhook URL and log the event. The token will not be automatically revoked.

Reviewer: ```suggestion
> **Note:** If the `revoke` option is disabled, the service only sends a notification to the configured webhook URL and logs the event. The token is not automatically revoked.
```

> ### Outgoing webhook

Reviewer: I feel like a heading here isn't necessary.

> ### Outgoing webhook

Reviewer: ```suggestion
```

> By default, when a token leak is found, the service will automatically revoke the token and log the event.

Reviewer: ```suggestion
By default, when the Grafana secret scanning service detects a leaked token, the service automatically revokes the token and logs the event.
```

Additionally, the service can be configured to send an outgoing webhook notification to a webhook URL.

> This notification will contain a JSON payload with the following data:

Reviewer: ```suggestion
The notification includes a JSON payload that contains the following data:
```

```json
{
  "alert_uid": "c9ce50a1-d66b-45e4-9b5d-175766cfc026",
  "link_to_upstream_details": <URL to token leak>,
  "message": "Token of type grafana_service_account_token with name
sa-the-toucans has been publicly exposed in <URL to token leak>.
Grafana has revoked this token",
  "state": "alerting",
  "title": "SecretScan Alert: Grafana Token leaked"
}
```

> ## Configuration

Reviewer: ```suggestion
## Steps
```

> > **Note:** The secretscan feature is disabled by default.

Reviewer: ```suggestion
> The secretscan feature is disabled by default.
```

> ```ini

Reviewer: Let's format this with numbered steps. 

1. Open the Grafana configuration file.

1. In the `[secretscan]`, update the following parameters:
